posthog/models/exchange_rate/sql.py

```diff
@@ -0,0 +1,166 @@
+#
+# The resulting CSV is stored in `historical.csv`
+#
+# This won't return values for dates where we didn't have a value.
+# When querying the table/dictionary, you'll need to look for the previous closest date with a value.
+def HISTORICAL_EXCHANGE_RATE_DICTIONARY():
+    rates_dict = {}
+    currencies = []
+
+    # Load the CSV file
+    with open(os.path.join(os.path.dirname(__file__), "historical.csv")) as f:
+        reader = csv.reader(f)
+
+        # Get header row with currency codes
+        currencies = next(reader)
+        currencies = [c.strip() for c in currencies]
+
+        # Parse each row
+        for row in reader:
+            if not row:  # Skip empty rows
+                continue
+
+            date_str = row[0].strip()
+            try:
+                # Parse the date
+                date = datetime.datetime.strptime(date_str, "%Y-%m-%d")
+                date_key = date.strftime("%Y-%m-%d")
+
+                # Create dictionary for this date
+                rates = {}
+                for i, value in enumerate(row[1:], 1):
+                    currency = currencies[i]
+
+                    # Only add non-empty values
+                    value = value.strip()
+                    if value:
+                        try:
+                            rates[currency] = float(value)
+                        except ValueError:
+                            # Just ignore non-numeric values
+                            pass
+
+                rates_dict[date_key] = rates
+            except ValueError:
+                # Skip rows with invalid dates
+                continue
+
+    # Fill in missing dates
+    start_date = datetime.datetime(2000, 1, 1)
+    end_date = datetime.datetime(2024, 12, 31)
+    current_date = start_date
+
+    while current_date <= end_date:
+        date_str = current_date.strftime("%Y-%m-%d")
+
+        if date_str not in rates_dict:
+            rates_dict[date_str] = {}
+
+        current_date += datetime.timedelta(days=1)
+
+    return rates_dict
+
+
+# Yield from HISTORICAL_EXCHANGE_RATE_DICTIONARY()
+# a tuple in the form of (date, currency, rate)
+def HISTORICAL_EXCHANGE_RATE_TUPLES():
+    rates_dict = HISTORICAL_EXCHANGE_RATE_DICTIONARY()
+    for date, rates in rates_dict.items():
+        for currency, rate in rates.items():
+            yield (date, currency, rate)
+
+
+EXCHANGE_RATE_TABLE_NAME = "exchange_rate"
+EXCHANGE_RATE_DICTIONARY_NAME = "exchange_rate_dict"
+
+# `version` is used to ensure the latest version is kept, see `MergeTreeEngine`
+EXCHANGE_RATE_TABLE_SQL = (
+    lambda on_cluster=True: """
+CREATE TABLE IF NOT EXISTS {table_name} {on_cluster_clause} (
+    date Date,
+    currency String,
+    rate Decimal64(4),
+    version UInt32 DEFAULT toUnixTimestamp(now())
+) ENGINE = {engine}
+ORDER BY (date, currency);
+""".format(
+        table_name=EXCHANGE_RATE_TABLE_NAME,
+        engine=MergeTreeEngine(
```

Comment:
I think you should use ReplacingMergeTree here with ver instead of replacing_col.